src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -386,7 +390,155 @@ public void renderLines(PoseStack matrices, Tesselator tessellator, BufferBuilde
             );
         }
     }
+    public static class RenderedPolyface extends RenderedShape<ShapeDispatcher.Polyface>
+    {
+        public RenderedPolyface(Minecraft client, ShapeDispatcher.ExpiringShape shape)
+        {
+            super(client, (ShapeDispatcher.Polyface)shape);
+        }
+        @Override
+        public void renderFaces(Tesselator tessellator, BufferBuilder bufferBuilder, double cx, double cy, double cz, float partialTick)
+        {       
+            if(shape.fa==0){return;}
+            if (shape.doublesided)
+                RenderSystem.disableCull();
+            else
+                RenderSystem.enableCull();
+            
+            RenderSystem.depthMask(false);
+            RenderSystem.depthFunc(516);
+            RenderSystem.enableBlend();
+            RenderSystem.blendFunc(GlStateManager.SourceFactor.ONE_MINUS_DST_ALPHA, GlStateManager.DestFactor.DST_ALPHA);
```

Comment:
should i use `blendFuncSeparate` here instead? 
Not Sure.

If no one makes a complaint, I won't change it.